### PR TITLE
feat: Implement Lesson Module

### PR DESCRIPTION
**Pull request by sammylina**
This PR introduces a new `Lesson` module to the application.

**Changes include:**
- **Schema Update:** Added a `Lesson` model to `prisma/schema.prisma` with fields for title, audio URL, duration, playlist ID, and order. A relationship between `Playlist` and `Lesson` has also been established.
- **Service Layer:** Created `src/modules/lesson/lesson.service.ts` with functions for creating, retrieving (by ID and playlist ID), updating, and deleting lessons.
- **Controller Layer:** Created `src/modules/lesson/lesson.controller.ts` with API endpoint handlers for lesson operations.
- **Routes:** Defined `src/modules/lesson/lesson.routes.ts` to map lesson API endpoints to their respective controller functions.
- **Module Integration:** Updated `src/modules/index.ts` and `src/routes/index.ts` to include and expose the new lesson routes.
- **Database Migration:** Applied Prisma migrations to update the database schema with the new `Lesson` model.

This module allows for the creation, management, and retrieval of individual audio lessons, which can be associated with playlists.

## Summary by CodeRabbit

* **New Features**
  * Added support for creating lessons within playlists, including a new API endpoint for lesson creation.
  * Playlists can now include multiple lessons, and lessons are associated with playlists.
  * Introduced a new endpoint to retrieve a playlist along with its lessons.

* **Bug Fixes**
  * Improved input validation and error handling for lesson and playlist operations.

* **Tests**
  * Added comprehensive tests for lesson creation, validation, and authorization.
  * Expanded test coverage for playlist operations, including edge cases and error handling.

* **Chores**
  * Updated dependency management for development tools.

### CODE REVIEW

**Comment by sammylina** [comment]
@coderabbitai 
I implemented a lesson feature based on issue #19, make sure we are not fetching lesson while during fetching playlists. we want to read the lessons if a specific playlist is requested

File: package.json [hunk 1]
```diff
@@ -69,6 +68,7 @@
     "lint-staged": "^16.1.2",
     "nodemon": "^3.1.10",
     "prettier": "^3.6.2",
+    "prisma": "^6.13.0",
```
[the remark this answers is not in the document]
Comment (sammylina) [reply package.json:71]: @coderabbitai apply thie recommendation
